web/src/js/ducks/modes/local.ts

```diff
@@ -3,20 +3,35 @@ import {
     UPDATE as UPDATE_STATE,
 } from "../backendState";
 import { addSetter, createModeUpdateThunk, updateState } from "./utils";
-import { createSlice } from "@reduxjs/toolkit";
-import { LocalState, parseRaw } from "../../modes/local";
+import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
+import { LocalState, parseRaw, Process } from "../../modes/local";
+import { fetchApi } from "../../utils";
 
 export const setActive = createModeUpdateThunk<boolean>(
     "modes/local/setActive",
 );
-export const setApplications = createModeUpdateThunk<string | undefined>(
-    "modes/local/setApplications",
+export const setSelectedProcesses = createModeUpdateThunk<string | undefined>(
+    "modes/local/setSelectedProcesses",
+);
+
+export const fetchProcesses = createAsyncThunk(
+    "modes/local/fetchProcesses",
+    async (_, { rejectWithValue }) => {
+        try {
+            const response = await fetchApi("/processes");
+            return response.json();
+        } catch (error) {
+            return rejectWithValue(error.message);
+        }
+    },
 );
 
 export const initialState: LocalState[] = [
     {
         active: false,
-        applications: "",
+        isLoading: false,
+        currentProcesses: [],
+        selectedProcesses: "",
```

Comment:
What do you think about moving the process list state into its own slice at the top level? Technically speaking it's completely separate from the mode spec. :) And then I'm not confused on why we have both `currentProcesses` and `selectedProcesses`. 😅